plugins/inputs/jti_openconfig_telemetry/openconfig_telemetry.go

```diff
@@ -374,8 +383,8 @@ func (m *OpenConfigTelemetry) Start(acc telegraf.Accumulator) error {
 		}
 
 		// If a certificate is provided, open a secure channel. Else open insecure one
-		if transportCredentials != nil {
-			grpcClientConn, err = grpc.Dial(server, grpc.WithTransportCredentials(transportCredentials))
+		if tlscfg != nil {
+			grpcClientConn, err = grpc.Dial(server, grpc.WithTransportCredentials(credentials.NewTLS(tlscfg)))
```

Comment:
All the code is right, but if you are interested in going further I can see a way we could have done this nicer originally. We should have created a `grpc.DialOption` above based on EnableTLS, and then here we don't need the if/else.  Optional.

Reply:
done